ui/packages/catalog-ui-search/src/main/webapp/component/query-advanced/query-advanced.view.js

```diff
@@ -194,7 +194,19 @@ module.exports = Marionette.LayoutView.extend({
     }
   },
   validate() {
-    return this.querySettings.currentView.validate()
+    const errorMessages = this.querySettings.currentView.validate()
+    const filters = this.queryAdvanced.currentView.getFilters().filters
+    for (let i = 0; i < filters.length; i++) {
+      const filter = filters[i]
+      const geometry = filter.geojson && filter.geojson.geometry ? filter.geojson.geometry : null
+      if (geometry && geometry.type === 'Polygon' && geometry.coordinates[0].length < 4) {
```

Comment:
I think this should be `< 3` instead

Reply:
Polygons have to have at least 3 points, with the fourth being a repeat of the first to close it up